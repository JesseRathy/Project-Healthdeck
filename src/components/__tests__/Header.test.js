import {Header} from '../index.js';
import { shallow,configure,mount } from 'enzyme';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
//Imports to get Enzyme to work if possible -> will work on later


test('renders correctly', () => {
  const wrapper = shallow(<Header headerTitle="This is a title"/>);
  expect(wrapper).toMatchSnapshot();
  //expect(tree).toMatchSnapshot();
});

/*this test currently does not work; note that this is due to requiring an actual DOM document to work from.
this shouldn't be difficult, but there's a lot of people saying that shallow renders are enough to make sure it renders properly
at least on the face of it. I will go into testing SPECIFIC pieces of components rendering properly for ID3 now that
I can actually write test code that actually tests things. */

/*it('Formats Title Correctly', () => {
  const wrapper = mount(
    <Header headerTitle="This is a title"/>
  );
  const text = wrapper.find("Text").text();

  expect(text).toEqual("This is a title");
});*/
