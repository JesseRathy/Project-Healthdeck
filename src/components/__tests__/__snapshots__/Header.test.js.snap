// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    headerTitle="This is a title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        This is a title
      </Component>,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#F8F8F8",
        "elevation": 2,
        "height": 60,
        "justifyContent": "center",
        "paddingTop": 15,
        "position": "relative",
        "shadowColor": "#000",
        "shadowOffSet": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.2,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "This is a title",
        "style": Object {
          "fontSize": 20,
        },
      },
      "ref": null,
      "rendered": "This is a title",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          This is a title
        </Component>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#F8F8F8",
          "elevation": 2,
          "height": 60,
          "justifyContent": "center",
          "paddingTop": 15,
          "position": "relative",
          "shadowColor": "#000",
          "shadowOffSet": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.2,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "This is a title",
          "style": Object {
            "fontSize": 20,
          },
        },
        "ref": null,
        "rendered": "This is a title",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
